### PR TITLE
Initial time range selection UX.

### PR DESCRIPTION
**Pull request by pmuetschard**

### CODE REVIEW

**Comment by rschiu** [comment]
LGTM functionality-wise.  Code looks reasonable, but probably not qualified to approve it.

Is it intentional that I need to press shift to highlight even though I already need to move the mouse to the timeline bar?  I am fine with it, just want to double check.

File: gapic/src/main/com/google/gapid/perfetto/canvas/Panel.java [hunk 1]
```diff
@@ -148,6 +149,36 @@ public boolean click() {
         }
       };
     }
+
+    public default Panel.Hover withClick(BooleanSupplier onClick) {
+      return new Hover() {
+        @Override
+        public Area getRedraw() {
```
Comment (pau-baiget) [inline gapic/src/main/com/google/gapid/perfetto/canvas/Panel.java]: [Possibly ignorant question]: I'm wondering, is the overridden function doing anything different?

File: gapic/src/main/com/google/gapid/perfetto/models/Selection.java [hunk 2]
```diff
@@ -58,6 +59,14 @@ public Composite buildUi(Composite parent, State state) {
         return new MultiSelectionView(parent, selections, state);
       }
     }
+
+    @Override
+    public void mark(State state) {
+      // TODO: is this good enough?
```
Comment (pau-baiget) [inline gapic/src/main/com/google/gapid/perfetto/models/Selection.java]: Looks good enough to me.

File: gapic/src/main/com/google/gapid/perfetto/canvas/Panel.java (hunk 1 above)
Comment (pmuetschard) [reply gapic/src/main/com/google/gapid/perfetto/canvas/Panel.java]: `withClick` creates a delegate that augments the `click()` function below. However, all other calls have to be propagated to the delegate.

**Pull request merged**